digest/digest_test.go

```diff
@@ -14,21 +12,6 @@ func TestParseDigest(t *testing.T) {
 		hex       string
 	}{
 		{
-			input:     "tarsum+sha256:e58fcf7418d4390dec8e8fb69d88c06ec07039d651fedd3aa72af9972e7d046b",
-			algorithm: "tarsum+sha256",
```

Comment:
We would need an algorithm that supports this in order to test it. The old code had special cases that allowed this syntax for tarsum. The `hash.Hash` based algorithms can't handle it.